client/network/src/service/out_events.rs

```diff
@@ -0,0 +1,267 @@
+		// we have sent an element on the channel.
+		if let Some(metrics) = &*self.metrics {
+			increase_metrics(metrics, &event, self.event_streams.len() as u64);
+		}
+	}
+}
+
+impl fmt::Debug for OutChannels {
+	fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+		f.debug_struct("OutChannels")
+			.field("num_channels", &self.event_streams.len())
+			.finish()
+	}
+}
+
+struct Metrics {
+	// This list is ordered alphabetically
+	out_events_dht_count: Gauge<U64>,
+	out_events_num_channels: Gauge<U64>,
+	out_events_notifications_closed_count: GaugeVec<U64>,
+	out_events_notifications_opened_count: GaugeVec<U64>,
+	out_events_notifications_sizes: GaugeVec<U64>,
+}
+
+impl Metrics {
+	fn register(registry: &Registry) -> Result<Self, PrometheusError> {
+		Ok(Self {
+			out_events_dht_count: register(Gauge::new(
+				"sub_libp2p_out_events_dht_count",
+				"Number of DHT events currently pending in the channels that broadcast network events",
+			)?, registry)?,
+			out_events_num_channels: register(Gauge::new(
+				"sub_libp2p_out_events_num_channels",
+				"Number of internal active channels that broadcast network events",
+			)?, registry)?,
+			out_events_notifications_closed_count: register(GaugeVec::new(
+				Opts::new(
+					"sub_libp2p_out_events_notifications_closed_count",
+					"Number of notification substreams opened events pending in the channels that broadcast network events"
+				),
+				&["protocol"]
+			)?, registry)?,
+			out_events_notifications_opened_count: register(GaugeVec::new(
+				Opts::new(
+					"sub_libp2p_out_events_notifications_opened_count",
+					"Number of notification substreams opened events pending in the channels that broadcast network events"
+				),
+				&["protocol"]
+			)?, registry)?,
+			out_events_notifications_sizes: register(GaugeVec::new(
+				Opts::new(
+					"sub_libp2p_out_events_notifications_sizes",
+					"Total size of notification events pending in the channels that broadcast network events"
+				),
+				&["protocol"]
+			)?, registry)?,
+		})
+	}
+}
+
+fn increase_metrics(metrics: &Metrics, event: &Event, num: u64) {
+	match event {
+		Event::Dht(_) => metrics.out_events_dht_count.inc(),
+		Event::NotificationStreamOpened { engine_id, .. } => {
+			metrics.out_events_notifications_opened_count
+				.with_label_values(&[&engine_id_to_string(engine_id)])
+				.add(num);
+		},
+		Event::NotificationStreamClosed { engine_id, .. } => {
+			metrics.out_events_notifications_closed_count
+				.with_label_values(&[&engine_id_to_string(engine_id)])
+				.add(num);
+		},
+		Event::NotificationsReceived { messages, .. } => {
+			for (engine_id, message) in messages {
+				metrics.out_events_notifications_sizes
+					.with_label_values(&[&engine_id_to_string(engine_id)])
+					.add(num.saturating_mul(u64::try_from(message.len()).unwrap_or(u64::max_value())));
+			}
+		},
+	}
+}
+
+fn decrease_metrics(metrics: &Metrics, event: &Event) {
```

Comment:
Decreasing metrics instead of monotonically increasing hides bursts.

Instead of using a Gauge to count the amount of elements in the channel, one could have a counter for sending and one for receiving. Subtracting the two still leads to the amount of items in the channel, but looking at each of them individually one can spot bursts.

The sending and receiving does not need to be realized as two counters, but one could just use a single counter introducing an `action` label with `send` and `received`.